interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java

```diff
@@ -0,0 +1,156 @@
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.testing.integration;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import io.grpc.ManagedChannel;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.testing.TestUtils;
+import io.grpc.testing.integration.Metrics.EmptyMessage;
+import io.grpc.testing.integration.Metrics.GaugeResponse;
+import io.grpc.testing.integration.StressTestClient.TestCaseWeightPair;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.net.InetSocketAddress;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.LockSupport;
+
+/** Unit tests for {@link StressTestClient}. */
+@RunWith(JUnit4.class)
+public class StressTestClientTest {
+
+  @Test
+  public void testDefaults() {
+    StressTestClient client = new StressTestClient();
+    assertEquals(singletonList(new InetSocketAddress("localhost", 8080)), client.addresses());
+    assertTrue(client.testCaseWeightPairs().isEmpty());
+    assertEquals(-1, client.durationSecs());
+    assertEquals(1, client.channelsPerServer());
+    assertEquals(1, client.stubsPerChannel());
+    assertEquals(8081, client.metricsPort());
+  }
+
+  @Test
+  public void allCommandlineSwitchesAreSupported() {
+    StressTestClient client = new StressTestClient();
+    client.parseArgs(new String[] {
+        "--server_addresses=localhost:8080,localhost:8081,localhost:8082",
+        "--test_cases=empty_unary:20,large_unary:50,server_streaming:30",
+        "--test_duration-secs=20",
```

Comment:
This is `test_duration_secs`  (i.e underscore between `duration` and `secs` instead of a dash). Sorry, the spec initially had a typo and I fixed it.